src/psyclone/tests/domain/nemo/transformations/nemo_outerarrayrange2loop_trans_test.py

```diff
@@ -98,6 +98,65 @@ def test_transform_apply_mixed_implicit_do(tmpdir):
     assert Compile(tmpdir).string_compiles(result)
 
 
+def test_apply_with_structures(fortran_reader, fortran_writer):
+    '''Check that the PSyIR is transformed as expected when the
+    expressions contain a mix of arrays and structures in the same
+    reference.
+
+    '''
+    trans = NemoOuterArrayRange2LoopTrans()
+
+    # The inner dimension is already set
+    psyir = fortran_reader.psyir_from_source('''
+    subroutine test
+        use my_variables
+        integer, parameter :: constant = 3
+        integer :: jk
+        base%field(constant)%array(:,:,jk) = 1
+    end subroutine test
+    ''')
+    assignment = psyir.walk(Assignment)[0]
+    trans.apply(assignment)
+    result = fortran_writer(assignment)
+    assert "base%field(constant)%array(:,jj,jk) = 1" in result
+    trans.apply(assignment)
+    result = fortran_writer(assignment)
+    assert "base%field(constant)%array(ji,jj,jk) = 1" in result
+
+    # The outer dimension is already set
+    psyir = fortran_reader.psyir_from_source('''
+    subroutine test
+        use my_variables
+        integer, parameter :: jf = 3, jpi = 3, jpim1 = 1
+        ptab(jf)%pt2d(jpi,:,:) = ptab(jf)%pt2d(jpim1,:,:)
+    end subroutine test
+    ''')
+    assignment = psyir.walk(Assignment)[0]
+    trans.apply(assignment)
+    result = fortran_writer(assignment)
+    assert "ptab(jf)%pt2d(jpi,:,jk) = ptab(jf)%pt2d(jpim1,:,jk)" in result
+    trans.apply(assignment)
+    result = fortran_writer(assignment)
+    assert "ptab(jf)%pt2d(jpi,jj,jk) = ptab(jf)%pt2d(jpim1,jj,jk)" in result
+
+    # Nested range structures are not supported
+    psyir = fortran_reader.psyir_from_source('''
+    subroutine test
+        use my_variables
+        integer, parameter :: jf = 3, jpi = 3, jpim1 = 1
+        ptab(:)%pt2d(jpi,:,:) = ptab(:)%pt2d(jpim1,:,:)
```

Comment:
You know what I'm going to say about this one... :-)

Reply:
Deleted